### PR TITLE
16636 Added new component "SubMenu"

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16636

**I needed something like this so I coded it. In the end, it wasn't exactly what I needed, but instead of throwing away the code, I thought I'd publish it here.**

*Description of changes:*
- commit 1: added new component
- commit 2: rebuilt Storybook

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
Sample:

image

File: src/components/sub-menu/SubMenu.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        {{ name }}
+      </v-btn>
+    </template>
+    <v-list>
+      <template v-for="(item, index) in menuItems">
+        <v-divider
+          v-if="item.isDivider"
+          :key="index"
+        />
+        <sub-menu
```
Comment (severinbeauvais) [inline src/components/sub-menu/SubMenu.vue:43]: Note the "recursion" here. This results in a submenu to the side of the current menu. What I actually wanted is a "nested" submenu, which opens in the same menu as the parent, like this:

image

File: src/components/sub-menu/SubMenu.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * Adapted from https://codepen.io/Moloth/pen/ZEBOzQP.
+ * See also https://github.com/vuetifyjs/vuetify/issues/1877.
+ */
+@Component({})
+export default class SubMenu extends Vue {
+  // props
+  @Prop() readonly name!: string
+  @Prop() readonly icon!: string
+  @Prop() readonly menuItems!: Array<any>
```
Comment (JazzarKarim) [inline src/components/sub-menu/SubMenu.vue:79]: It's not possible here to specify the type of the Array right Sev? I'm looking and I'm seeing the array is of mixed things.

**Review by JazzarKarim — APPROVED** [review]
LGTM Sev 👍

**Review by ketaki-deodhar — APPROVED** [review]
LGTM!

File: src/components/sub-menu/SubMenu.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/sub-menu/SubMenu.vue:79]: Yes, in this scenario the array is mixed items. One could come up with an interface (possibly a union of interfaces) that would work here, but I don't want to spend any more time on this as I won't be using it.

File: src/components/sub-menu/SubMenu.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/sub-menu/SubMenu.vue:43]: I have now created a generic component that does the above ^^ and which I will reuse for Home Jurisdictions. You'll see it in a future PR :)

File: src/components/sub-menu/SubMenu.vue (hunk 2 above)
Comment (JazzarKarim) [reply src/components/sub-menu/SubMenu.vue:79]: Totally agree. I was just out asking out of curiosity. Alright Sev, resolving.

**Comment by severinbeauvais** [comment]
image

**Pull request merged**